ocf_datapipes/select/select_spatial_slice.py

```diff
@@ -21,7 +21,249 @@
 logger = logging.getLogger(__name__)
 
 
-def select_spatial_slice_pixels(
+# -------------------------------- utility functions --------------------------------
+
+
+def convert_coords_to_match_xarray(x, y, from_coords, xr_data):
+    """Convert x and y coords to cooridnate system matching xarray data
+
+    Args:
+        x: Float or array-like
+        y: Float or array-like
+        from_coords: String describing coordinate system of x and y
+        xr_data: xarray data object to which coordinates should be matched
+    """
+
+    xr_coords, xr_x_dim, xr_y_dim = spatial_coord_type(xr_data)
+
+    assert from_coords in ["osgb", "lon_lat"]
+    assert xr_coords in ["geostationary", "osgb", "lon_lat"]
+
+    if xr_coords == "geostationary":
+        if from_coords == "osgb":
+            x, y = osgb_to_geostationary_area_coords(x, y, xr_data)
+
+        elif from_coords == "lon_lat":
+            x, y = lon_lat_to_geostationary_area_coords(x, y, xr_data)
+
+    elif xr_coords == "lon_lat":
+        if from_coords == "osgb":
+            x, y = osgb_to_lon_lat(x, y)
+
+        # else the from_coords=="lon_lat" and we don't need to convert
+
+    elif xr_coords == "osgb":
+        if from_coords == "lon_lat":
+            x, y = lon_lat_to_osgb(x, y)
+
+        # else the from_coords=="osgb" and we don't need to convert
```

Comment:
if there worth putting and error message, if we dont have the message. or a logger statement if no converstaion is needed